### PR TITLE
feat: new interfaceTypeIdMatch rule

### PR DESCRIPTION
**Pull request by Jermayy**
solves [#483](https://github.com/gajus/eslint-plugin-flowtype/issues/483#issue-928700940)

cc: @gajus @angelica-bocanegra @Brianzchen

### CODE REVIEW

File: .README/README.md [hunk 1]
```diff
@@ -54,6 +54,10 @@ npm install eslint babel-eslint eslint-plugin-flowtype --save-dev
       2,
       "never"
     ],
+    "flowtype/interface-type-id-match": [
+      2,
+      "^([A-Z][a-z0-9]+)+Type$"
+    ],
```
Comment (Brianzchen) [inline .README/README.md:60]: Don't think is necessary, but fine to keep.

File: src/rules/interfaceTypeIdMatch.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+];
+
+const create = (context) => {
+  const pattern = new RegExp(context.options[0] || '^([A-Z][a-z0-9]*)+Type$');
+
+  const checkType = (interfaceDeclarationNode) => {
+    const typeIdentifierName = interfaceDeclarationNode.id.name;
+
+    if (!pattern.test(typeIdentifierName)) {
+      context.report(interfaceDeclarationNode, 'Type identifier \'{{name}}\' does not match pattern \'{{pattern}}\'.', {
```
Comment (Brianzchen) [inline src/rules/interfaceTypeIdMatch.js:14]:
```suggestion
      context.report(interfaceDeclarationNode, 'Interface identifier \'{{name}}\' does not match pattern \'{{pattern}}\'.', {
```

File: src/rules/interfaceTypeIdMatch.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  {
+    type: 'string',
+  },
+];
+
+const create = (context) => {
+  const pattern = new RegExp(context.options[0] || '^([A-Z][a-z0-9]*)+Type$');
+
+  const checkType = (interfaceDeclarationNode) => {
+    const typeIdentifierName = interfaceDeclarationNode.id.name;
```
Comment (Brianzchen) [inline src/rules/interfaceTypeIdMatch.js:11]:
```suggestion
    const interfaceIdentifierName = interfaceDeclarationNode.id.name;
```

File: src/rules/interfaceTypeIdMatch.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
+const schema = [
+  {
+    type: 'string',
+  },
+];
+
+const create = (context) => {
+  const pattern = new RegExp(context.options[0] || '^([A-Z][a-z0-9]*)+Type$');
+
+  const checkType = (interfaceDeclarationNode) => {
```
Comment (Brianzchen) [inline src/rules/interfaceTypeIdMatch.js:10]:
```suggestion
  const checkInterface = (interfaceDeclarationNode) => {
```

File: tests/rules/index.js [hunk 5]
```diff
@@ -19,6 +19,7 @@ const reportingRules = [
   'delimiter-dangle',
   'enforce-line-break',
   'generic-spacing',
+  'interface-type-id-match',
```
Comment (Brianzchen) [inline tests/rules/index.js:22]:
```suggestion
  'interface-id-match',
```

**Review by Brianzchen — COMMENTED** [review]
Just left some suggestions on cleaning up the naming conventions because I know you copied this from type id match, it should ideally reference more to interface id match in the code and rule name

**Review by Brianzchen — APPROVED** [review]
LGTM

**Comment by Jermayy** [comment]
closing in favour of #492

**Pull request closed without merging**